src/main/java/seedu/address/logic/commands/RemarkCommand.java

```diff
@@ -0,0 +1,125 @@
+import seedu.address.model.person.Person;
+import seedu.address.model.person.ReadOnlyPerson;
+import seedu.address.model.person.Remark;
+import seedu.address.model.person.exceptions.DuplicatePersonException;
+import seedu.address.model.person.exceptions.PersonNotFoundException;
+
+/**
+ * Edits the remark of a person to the address book.
+ */
+public class RemarkCommand extends UndoableCommand {
+
+    public static final String COMMAND_WORD = "remark";
+    public static final String COMMAND_ALIAS = "rm";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + ": If remarks the person identified by the index number used in the last person listing,"
+            + " add the remark to the person.\n"
+            + "If the remark field is empty, the remark is removed for the person.\n"
+            + "Parameters: INDEX (must be a positive integer) "
+            + PREFIX_REMARK + "[REMARK] \n"
+            + "Example: (add remark) " + COMMAND_WORD + " 1 " + PREFIX_REMARK + "Likes to drink coffee.\n"
+            + "Example: (delete remark) " + COMMAND_WORD
+            + " 2 "
+            + PREFIX_REMARK + "\n";
+
+    public static final String MESSAGE_ADD_REMARK_SUCCESS = "Added Remark to Person: %1$s";
+    public static final String MESSAGE_DELETE_REMARK_SUCCESS = "Removed Remark from Person: %1$s";
+    public static final String MESSAGE_DUPLICATE_PERSON = "This person already exists in the address book.";
+
+    private final Index targetIndex;
+    private final Remark remark;
+
+    /**
+     * @param targetIndex of the person in the list to edit the remark
+     * @param remark of the person
+     */
+    public RemarkCommand(Index targetIndex, Remark remark) {
+        requireNonNull(targetIndex);
+        requireNonNull(remark);
+
+        this.targetIndex = targetIndex;
+        this.remark = remark;
+    }
+
+
+    @Override
+    public CommandResult executeUndoableCommand() throws CommandException {
+
+        List<ReadOnlyPerson> lastShownList = model.getFilteredPersonList();
+
+        if (targetIndex.getZeroBased() >= lastShownList.size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
+        }
+
+        ReadOnlyPerson personToRemark = lastShownList.get(targetIndex.getZeroBased());
+
+        Person remarkedPerson = createRemarkedPerson(personToRemark, remark);
+
+        try {
+            model.updatePerson(personToRemark, remarkedPerson);
+        } catch (DuplicatePersonException dpe) {
+            throw new CommandException(MESSAGE_DUPLICATE_PERSON);
+        } catch (PersonNotFoundException pnfe) {
+            throw new AssertionError("The target person cannot be missing");
+        }
+        model.updateFilteredPersonList(PREDICATE_SHOW_ALL_PERSONS);
+
+        return new CommandResult(generateSuccessMsg(remarkedPerson));
+    }
+
+    /**
+     * Generate the Successful Message accordingly.
```

Comment:
Coding standard violation:exclamation: The first statement of the header comment does not follow the prescribed style. See point 4, 3rd bullet point of the [[Java Coding Standard: Comments](https://oss-generic.github.io/process/codingStandards/CodingStandard-Java.html#comments)]

Reply:
@April0616 reminder if you have addressed this

Reply:
thanks !